vortex-expr/src/transform/remove_select.rs

```diff
@@ -4,8 +4,7 @@ use vortex_error::{vortex_err, VortexResult};
 use crate::traversal::{MutNodeVisitor, Node, TransformResult};
 use crate::{get_item, pack, ExprRef, Select};
 
-/// Select is a useful expression, however it can be defined in terms of get_item & pack,
-/// once the expression type is known, this simplifications pass removes the select expression.
+/// Replaces [Select] with combination of [GetItem] and [Pack] expressions.
 pub fn remove_select(e: ExprRef, scope_dt: &DType) -> VortexResult<ExprRef> {
```

Comment:
I think its worth saying that the expr is now bound to this type. [maybe, out of scope] we want to explicitly do this?

Reply:
Could you update the comments accordingly

Reply:
Added a note in simplify typed